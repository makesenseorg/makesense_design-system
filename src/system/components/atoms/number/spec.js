import { shallowMount } from "@vue/test-utils";
import Comp from "./Number.vue";

describe("Number.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("matches snapshot with slot value number", () => {
    const wrapper = shallowMount(Comp, {
      slots: {
        value: "2000",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("matches snapshot with slots", () => {
    const wrapper = shallowMount(Comp, {
      slots: {
        value: "2000",
        label: "Visitors",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("sets color modifier", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        color: "primary",
      },
    });

    expect(wrapper.classes()).toContain("number--color-primary");
  });
});
