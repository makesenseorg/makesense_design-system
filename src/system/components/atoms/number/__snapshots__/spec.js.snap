// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number.vue matches snapshot 1`] = `
<div
  class="number number--color-undefined"
>
  <div
    class="number__value"
  />
   
  <div
    class="number__label"
  />
</div>
`;

exports[`Number.vue matches snapshot with slot value number 1`] = `
<div
  class="number number--color-undefined"
>
  <div
    class="number__value"
  >
    2000
  </div>
   
  <div
    class="number__label"
  />
</div>
`;

exports[`Number.vue matches snapshot with slots 1`] = `
<div
  class="number number--color-undefined"
>
  <div
    class="number__value"
  >
    2000
  </div>
   
  <div
    class="number__label"
  >
    Visitors
  </div>
</div>
`;
